redis/src/main/scala/zio/redis/Output.scala

```diff
@@ -199,10 +200,267 @@ object Output {
+      var pos    = start
+      val output = Array.ofDim[GeoView](len)
+
+      while (idx < len) {
+        val member = unsafeReadString(text, pos)
+
+        while (text.charAt(pos) != '\n') pos += 1
+        pos += 1
+
+        val containsDistance = text.charAt(pos) != ':'
+        val longLat          = if (coordinates.isEmpty) None else Some(coordinates(idx))
+
+        if (!containsDistance) {
+          pos += 1
+          output(idx) = GeoView(member, None, Some(unsafeReadString(text, pos).toLong), longLat)
+        } else {
+          while (text.charAt(pos) != '\n') pos += 1
+          pos += 1
+
+          val distance = unsafeReadString(text, pos).toDouble
+
+          while (text.charAt(pos) != ':') pos += 1
+          pos += 1
+
+          output(idx) = GeoView(member, Some(distance), Some(unsafeReadString(text, pos).toLong), longLat)
+        }
+
+        idx += 1
+        if (idx < len) while (!text.charAt(pos).isLetter) pos += 1
+      }
+
+      output
+    }
+
+    private[this] def geoViewWithoutHash(
+      text: String,
+      start: Int,
+      len: Int,
+      coordinates: Chunk[LongLat]
+    ): Array[GeoView] = {
+      var idx    = 0
+      var pos    = start
+      val output = Array.ofDim[GeoView](len)
+
+      while (idx < len) {
+        val member           = unsafeReadString(text, pos)
+        while (text.charAt(pos) != '\n') pos += 1
+        pos += 1
+        val containsDistance = text.charAt(pos) == '$'
+        if (containsDistance) {
+          while (text.charAt(pos) != '\n') pos += 1
+          pos += 1
+          output(idx) = GeoView(member, Some(unsafeReadString(text, pos).toDouble), None, Some(coordinates(idx)))
+        } else output(idx) = GeoView(member, None, None, Some(coordinates(idx)))
+
+        idx += 1
+        if (idx < len) while (!text.charAt(pos).isLetter) pos += 1
+      }
+
+      output
+    }
+
+    private[this] def unsafeReadMembers(text: String, len: Int): Chunk[GeoView] = {
+      var idx    = 0
+      var pos    = 0
+      val output = Array.ofDim[GeoView](len)
+
+      while (idx < len) {
+        if (text.charAt(pos) == '$') {
+          while (text.charAt(pos) != '\n') pos += 1
+          pos += 1
+
+          output(idx) = GeoView(unsafeReadString(text, pos), None, None, None)
+          idx += 1
+        }
+        pos += 1
+      }
+
+      Chunk.fromArray(output)
+    }
+
+    private[this] def unsafeReadString(text: String, start: Int): String = {
+      var pos = start + 1
+
+      while (text.charAt(pos) != '\r') pos += 1
+
+      text.substring(start, pos)
+    }
+
+    private[this] def skipToNext(text: String, start: Int): Int = {
+      var pos = start
+
+      while (text.charAt(pos) != '$') pos += 1
+      while (text.charAt(pos) != '\n') pos += 1
+      pos += 1
+
+      pos
+    }
+  }
+
+  private[this] def unsafeReadLength(text: String): Int = {
+    var pos = 1
+
+    while (text.charAt(pos) != '\r') pos += 1
+
+    text.substring(1, pos).toInt
```

Comment:
This is a duplicate of `unsafeReadNumber`, therefore drop it. Also, try replacing the body of `unsafeReadNumber` with "manual" approach:

```scala
var pos = 1
var res = 0

while (text.charAt(pos) != '\r') {
  res = res * 10 + text.charAt(pos) - '0'
  pos += 1
}

res
```

If possible, find similar patterns and replace them with the call to this function.